### PR TITLE
Dbp 589 standardize naming conventions and structure in dbildungs iam keycloak helm chart

### PR DESCRIPTION
**Pull request by sahassou**
# Description
The aim is to address inconsistencies in naming conventions across our Helm charts in [dBildungsplattform/dbildungs-iam-keycloak (github.com)](https://github.com/dBildungsplattform/dbildungs-iam-keycloak/tree/main). The goal is to enhance clarity through:

Standardize Naming Conventions
Uniform values.yml Structure

TODO:

Conduct a review of all existing Helm charts to identify variations in naming conventions.
Develop a guideline for standardized naming conventions and values.yml file structure.
Perform an analysis of lookups usage in current Helm charts and recommend improvements or alternatives.
Evaluate the current implementation of chart dependencies
AC:

Naming conventions and structure in Helm charts are harmonized

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by kristoff-kiefer — APPROVED** [review]

File: charts/dbildungscloud-iam-keycloak/templates/configmap.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{{if not (lookup "v1" "ConfigMap" .Release.Namespace .Values.configmap.name)}}
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ .Values.configmap.name }}
-data:  
-  db-name: "{{.Values.db.name}}"  
-{{end}}
\ No newline at end of file
+  name: {{ template "common.names.configmap" . }}
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/templates/configmap.yaml:4]: Please add a namespace Parameter

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml [hunk 2]
```diff
@@ -1,102 +1,101 @@
 apiVersion: apps/v1
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:1]: Please add all Security relevant inputs. The helm scanner wants e.g. :  CPU and Memory Limits and Requests. NET_RAW Capabilities Not Being Dropped. Seccomp Profile Is Not Configured. Service Account Token Automount Not Disabled.

File: charts/dbildungscloud-iam-keycloak/templates/ingress.yaml [hunk 3]
```diff
@@ -1,20 +1,22 @@
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
-  name: keycloak-ingress
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/templates/ingress.yaml:4]: Please add a namespace.

File: charts/dbildungscloud-iam-keycloak/templates/service.yaml [hunk 4]
```diff
@@ -1,15 +1,15 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: keycloak
-  labels:
-    app.kubernetes.io/name: dbildungs-iam
+  name: {{ template "common.names.service" . }}
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/templates/service.yaml:4]: please add a namespace

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 5]
```diff
@@ -1,11 +1,40 @@
-keycloakHostname: "keycloak.dev.spsh.dbildungsplattform.de"
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/values.yaml:1]: Changing the helm-value names here means that you also have to change it in the deploy pipeline in spsh-app-deploy. These variables (e.g. keycloakHostname, frontendHostname and backendHostname, db.name) are overwritten in the pipeline and must match.

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 6]
```diff
@@ -1,11 +1,40 @@
...
-  name: keycloak
+replicaCount: 1
 image:
   repository: ghcr.io/dbildungsplattform/dbildungs-iam
   tag: ""
-db:
+  imagepullpolicy: IfNotPresent
+  allowPrivilegeEscalation: false
+
+database:
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/values.yaml:8]: same for this variable.

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 7]
```diff
@@ -1,11 +1,40 @@
...
   name: ""
-frontendHostname: ""
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/values.yaml:11]: and this

**Review by M-Schiborr — CHANGES_REQUESTED** [review]

File: charts/dbildungscloud-iam-keycloak/templates/ingress.yaml (hunk 3 above)
Comment (kristoff-kiefer) [reply charts/dbildungscloud-iam-keycloak/templates/ingress.yaml:4]: More precisely, .Release.Namespace, however I seem to recall that helm actually does this on its own?

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml [hunk 8]
```diff
@@ -1,102 +1,101 @@
...
                   key: keycloak-clientSecret
             - name: KC_ADMIN_SECRET
               valueFrom:
                 secretKeyRef:
-                  name: {{ .Values.secrets.name }}
+                  name: {{ .Values.secrets.keycloaksecretname }}
                   key: keycloak-adminSecret
             - name: KC_ROOT_URL
-              value: "https://{{.Values.frontendHostname}}"
+              value: "https://{{ .Values.ingress.hostname }}"
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:93]: I think ingress.hostname is not correkt here, frontendHostname is not equal keycloakhostname

Comment (kristoff-kiefer) [reply charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:93]: No it isn't. Sorry I missed that. This must be frontendHostname it gets injected into the Keycloak-Config

**Review by kristoff-kiefer — APPROVED** [review]

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 9]
```diff
@@ -1,11 +1,86 @@
...
+readinessProbe:
+  httpGet:
+    path: "/health/ready"
+    port: web
+
+ingress:
+  ingressClassName: nginx
+  pathType: Prefix
+  keycloakHostname: "keycloak.dev.spsh.dbildungsplattform.de"
+  frontendHostname: ""
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/values.yaml:75]: That is not an Ingress variable

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 10]
```diff
@@ -1,11 +1,86 @@
...
+
+volumeConfig:
+  name: realm-config
+  mountPath: "/opt/keycloak/data/import"
+  readOnly: true
+
+volumePermissions:
+  enabled: true
+  resources:
+    limits: { memory: "128Mi", cpu: "500m" }
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/values.yaml:48]: Is that enough?

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml [hunk 11]
```diff
@@ -93,11 +99,11 @@ spec:
             - name: KC_ROOT_URL
               value: "https://{{.Values.frontendHostname}}"
```
Comment (M-Schiborr) [inline charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:100]: .Values.frontendHostname does not exist anymore

File: charts/dbildungscloud-iam-keycloak/values.yaml [hunk 12]
```diff
@@ -1,11 +1,88 @@
...
+
+volumeConfig:
+  name: realm-config
+  mountPath: "/opt/keycloak/data/import"
+  readOnly: true
+
+volumePermissions:
+  enabled: true
+  resources:
+    limits: { memory: "1Gi", cpu: "500m" }
```
Comment (dimapin) [inline charts/dbildungscloud-iam-keycloak/values.yaml:50]: Please use intended extra lines for memory and cpu

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml [hunk 13]
```diff
@@ -1,50 +1,57 @@
...
+  namespace: {{ .Values.namespace }}
   labels:
-    app.kubernetes.io/name: dbildungs-iam
+    app.kubernetes.io/name: {{ .Values.commonLabels.name }}
 spec:
   selector:
     matchLabels:
-      layer: dbildungs-iam-keycloak
-  replicas: 1
+      app.kubernetes.io/layer: {{ .Values.selector.layer }}
```
Comment (dimapin) [inline charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:11]: What is this label for and why is a selector defined in the deployment?

**Review by dimapin — CHANGES_REQUESTED** [review]

File: charts/dbildungscloud-iam-keycloak/templates/deployment.yaml (hunk 13 above)
Comment (sahassou) [reply charts/dbildungscloud-iam-keycloak/templates/deployment.yaml:11]: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

**Review by dimapin — APPROVED** [review]

**Review by dimapin — APPROVED** [review]

**Review by M-Schiborr — APPROVED** [review]

**Pull request merged**